lib/FrontendTool/FrontendTool.cpp

```diff
@@ -592,26 +594,33 @@ static bool performCompile(std::unique_ptr<CompilerInstance> &Instance,
   if (Invocation.getSILOptions().LinkMode == SILOptions::LinkAll)
     performSILLinking(SM.get(), true);
 
-  {
-    SharedTimer timer("SIL verification (pre-optimization)");
-    SM->verify();
+  if (!PrimarySourceFile) {
+    SM->linkAll();
+    SM->dump();
```

Comment:
The Onone optimization passes *are* optional. (I'm not talking about the mandatory passes; this is the specific set of "optimizations run even at Onone".)